components/sock_utils/src/getnameinfo.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <fcntl.h>
+#include "lwip/sockets.h"
+#include "lwip/netdb.h"
+#include "ifaddrs.h"
+#include "esp_check.h"
+
+int getnameinfo(const struct sockaddr *addr, socklen_t addrlen,
```

Comment:
Is there any constraints to support `sockaddr_in6` and family `AF_INET6`?

Reply:
Yes, this is the limitation of most functions here, as documented in the README